src/hooks/useAsyncFetch.js

```diff
@@ -0,0 +1,18 @@
+import { useVideos } from "../context/video-context";
+import {useEffect} from "react";
+import axios from "axios";
+
+export function useAsyncFetch({url, dispatchType, dispatchPayload}){
+    const {dispatch}  = useVideos();
+
+    useEffect(()=>{
+        (async function asyncFetch(){
+            try{
+                const {data} = await axios.get(url);
+                dispatch({type:dispatchType, payload: data[dispatchPayload]});
+            }catch(err){
+                console.log(err);
```

Comment:
console.log statements can be removed.

Reply:
Currently I'm not using toast or showing error on UI, therefore I have used console.log. Will remove it soon.